@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1709,8 +1727,14 @@ export class TextElementsRenderer {
         }
         // ... and render the icon (if any).
         if (iconReady) {
```

Comment:
```suggestion
        if (iconReady && !iconInvisible) {
```
that way you don't need to handle invisible icons below